### DISCUSSION TITLE
Is `poetry check` only for the abstract structure to the `pyproject.toml`?

### DISCUSSION THREAD

**Discussion by YDX-2147483647** [python-poetry/poetry#7424] [General] [answered]
Is [`poetry check`](https://python-poetry.org/docs/cli#check) only for the _abstract_ structure to the `pyproject.toml`?

To be specific, should this command check _existence of the file_ `docs/README.md` for the following `pyproject.toml`?

```toml
[tool.poetry]
# ...
readme = "docs/README.md"
```

---

(I'm trying to fix [#6937 `poetry install` checks `README.md`, but `poetry check` does not](https://github.com/python-poetry/poetry/issues/6937).)

**Comment by radoering** [comment, marked as answer]
The [documentation](https://python-poetry.org/docs/cli/#check) says that it validates the structure. `poetry check --help` says that it checks the validity of pyproject.toml. However, (maybe via flags) it should just check anything we want to check. We only have to think about the default and update the documentation. Checking the existence of referenced files makes sense.

Kind of related: It is planned to replace `poetry lock --check` with `poetry check --lock` (see #6756, `poetry lock --check` is considered a UI design flaw).

**Reply by YDX-2147483647** [reply]
Thanks! I’ll continue my work.
